crates/ide-db/src/imports/import_assets.rs

```diff
@@ -536,6 +537,29 @@ fn trait_applicable_items(
         required_assoc_items.insert(assoc);
         Some(assoc_item_trait.into())
     })
+    .filter(|candidate_trait_id| {
+        // blanket `Trait` implementations for type `A` can only exist in the crate defining
+        // the trait. However, an implementation for `A`` can only exist in `A`'s or `Trait`'s crate,
+        // which allows us to reduce the search space substantially.
```

Comment:
As I later pointer out on zulip, this is not correct for fundamental types. Example (please add something similar as a test):

```rust
struct A;

impl fmt::Display for Box<A> { ... }
impl fmt::Display for &A { ... }
```
Both of those impls [compile](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4e9622b393ae5e8aabbb93133e169703) even though the impls are not in `Display` or `Box`/`&`'s definition crate.

Similarly: 
```rust
struct A;

impl Into<A> for () {
    fn into(self) -> A { A }
}
```
Is also valid, even though both `Into` and unit are foreign.

You need to go through all generics.

Reply:
Note though that `impl fmt::Display for Vec<A> {  }` does **not** compile. So the crates to search for impls are
1. Trait's definition crate
2. Definition crates for all trait's generic arguments
    - This is recursive for _fundamental types_ — `Into<Box<A>> for ()` is OK, but `Into<Vec<A>> for ()` is NOT
3. Self type definition crate
     - This is recursive for _fundamental types_

Reply:
i'l add add a test!

Reply:
Wrote my own trait (`pub trait TestInto<T> { fn into(self) -> T; }`, following your suggestion on Zulip) and completions on `()` showed up.